framework/src/util/Utils.cpp

```diff
@@ -154,6 +154,13 @@ std::string GetCurrentWorkingDirectory()
   return std::string();
 }
 
+static const std::string s_CurrWorkingDir = InitCurrentWorkingDirectory();
```

Comment:
I am not sure why you think it is about `InitCurrentWorkingDirectory`. We need to ensure that we do not call `GetCurrentDirectory` in an unsafe way (provided by kernel32.dll). We cannot control who calls that function and when or in what thread it is called during a progams lifetime. How can a magic function local static help here? I fail to see that.